src/storage-client/src/storage_collections.rs

```diff
@@ -1044,6 +1161,87 @@ where
         Ok(())
     }
 
+    async fn allow_writes(&self) {
+        // Because of the limitations that we explain below, around how we can
+        // turn a leased ReadHandle into a SinceHandle.
+        //
+        // This will at least make sure that someone can notice this on Sentry
+        // if we ever forget that we're not meant to use this code path.
+        tracing::error!("coming out of read-only mode gracefully, in-process, is not meant to be used in production");
```

Comment:
Where is this used if it's not meant to be used in production?

Reply:
I started a slack discussion about this because both you and Jan remarked on this 😅